frontend/awx/resources/templates/TemplatePage/steps/CredentialPasswordsStep.tsx

```diff
@@ -0,0 +1,134 @@
+            showCredentialPasswordPrivilegeEscalation = true;
+          }
+          if (launchConfigCredential.passwords_needed.includes('ssh_key_unlock')) {
+            showCredentialPasswordPrivateKeyPassphrase = true;
+          }
+
+          const vaultPasswordIds = launchConfigCredential.passwords_needed
+            .filter((passwordNeeded) => passwordNeeded.startsWith('vault_password'))
+            .map((vaultPassword) => vaultPassword.split(/\.(.+)/)[1] || '');
+
+          vaultsThatPrompt.push(...vaultPasswordIds);
+        }
+      } else {
+        if (credential?.inputs?.password === 'ASK') {
+          showCredentialPasswordSsh = true;
+        }
+
+        if (credential?.inputs?.become_password === 'ASK') {
+          showCredentialPasswordPrivilegeEscalation = true;
+        }
+
+        if (credential?.inputs?.ssh_key_unlock === 'ASK') {
+          showCredentialPasswordPrivateKeyPassphrase = true;
+        }
+
+        if (credential?.inputs?.vault_password === 'ASK') {
+          vaultsThatPrompt.push(credential.inputs.vault_id.toString());
+        }
+      }
+    });
+  }
+
+  return (
+    <PageFormGrid isVertical singleColumn>
+      <ConditionalField isHidden={!showCredentialPasswordSsh}>
+        <PageFormTextInput<TemplateLaunch>
+          id="launch-ssh-password"
+          label={t('SSH password')}
+          name="credential_passwords.ssh_password"
+          placeholder={t('Enter a password')}
+          type="password"
+          isRequired
+        />
+      </ConditionalField>
+      <ConditionalField isHidden={!showCredentialPasswordPrivateKeyPassphrase}>
+        <PageFormTextInput<TemplateLaunch>
+          id="launch-private-key-passphrase"
+          label={t('Private key password')}
+          name="credential_passwords.ssh_key_unlock"
+          placeholder={t('Enter a password')}
+          type="password"
+          isRequired
+        />
+      </ConditionalField>
+      <ConditionalField isHidden={!showCredentialPasswordPrivilegeEscalation}>
+        <PageFormTextInput<TemplateLaunch>
+          id="launch-privilege-escalation-password"
+          label={t`Privilege escalation password`}
+          name="credential_passwords.become_password"
+          placeholder={t('Enter a password')}
+          type="password"
+          isRequired
+        />
+      </ConditionalField>
+      {vaultsThatPrompt.map((credId) => {
+        const passwordKey = `vault_password${credId !== '' ? `.${credId}` : ''}`;
+        return (
+          <PageFormTextInput<TemplateLaunch>
+            key={credId}
+            id={`launch-vault-password-${credId}`}
+            label={credId === '' ? t`Vault password` : t`Vault password | ${credId}`}
+            name={`credential_passwords.${passwordKey}`}
+            placeholder={t('Enter a password')}
+            type="password"
+            isRequired
+          />
+        );
+      })}
+    </PageFormGrid>
+  );
+}
+
+export function ConditionalField({
```

Comment:
We seem to be defining this component in 2 separate files.  Could we define it in 1 and import it where needed?

Reply:
Pushed an update that moves this component into its own separate file.